homeassistant/components/cover/group.py

```diff
@@ -0,0 +1,268 @@
+})
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the Group Cover platform."""
+    async_add_devices(
+        [CoverGroup(config[CONF_NAME], config[CONF_ENTITIES])])
+
+
+class CoverGroup(CoverDevice):
+    """Representation of a CoverGroup."""
+
+    def __init__(self, name, entities):
+        """Initialize a CoverGroup entity."""
+        self._name = name
+        self._is_closed = False
+        self._cover_position = 100
+        self._tilt_position = None
+        self._supported_features = 0
+        self._assumed_state = True
+
+        self._entities = entities
+        self._covers = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                        KEY_POSITION: set()}
+        self._tilts = {KEY_OPEN_CLOSE: set(), KEY_STOP: set(),
+                       KEY_POSITION: set()}
+
+    @callback
+    def update_supported_features(self, entity_id, old_state, new_state,
+                                  update_state=True):
+        """Update dictionaries with supported features."""
+        if new_state is None:
+            for values in self._covers.values():
+                values.discard(entity_id)
+            for values in self._tilts.values():
+                values.discard(entity_id)
+        else:
+            features = new_state.attributes.get(ATTR_SUPPORTED_FEATURES, 0)
+
+            if features & (SUPPORT_OPEN | SUPPORT_CLOSE):
+                self._covers[KEY_OPEN_CLOSE].add(entity_id)
+            else:
+                self._covers[KEY_OPEN_CLOSE].discard(entity_id)
+            if features & (SUPPORT_STOP):
+                self._covers[KEY_STOP].add(entity_id)
+            else:
+                self._covers[KEY_STOP].discard(entity_id)
+            if features & (SUPPORT_SET_POSITION):
+                self._covers[KEY_POSITION].add(entity_id)
+            else:
+                self._covers[KEY_POSITION].discard(entity_id)
+
+            if features & (SUPPORT_OPEN_TILT | SUPPORT_CLOSE_TILT):
+                self._tilts[KEY_OPEN_CLOSE].add(entity_id)
+            else:
+                self._tilts[KEY_OPEN_CLOSE].discard(entity_id)
+            if features & (SUPPORT_STOP_TILT):
+                self._tilts[KEY_STOP].add(entity_id)
+            else:
+                self._tilts[KEY_STOP].discard(entity_id)
+            if features & (SUPPORT_SET_TILT_POSITION):
+                self._tilts[KEY_POSITION].add(entity_id)
+            else:
+                self._tilts[KEY_POSITION].discard(entity_id)
+
+        if update_state:
+            self.async_schedule_update_ha_state(True)
+
+    async def async_added_to_hass(self):
+        """Register listeners."""
+        for entity_id in self._entities:
+            new_state = self.hass.states.get(entity_id)
+            self.update_supported_features(entity_id, None, new_state,
+                                           update_state=False)
+        await self.async_update()
```

Comment:
Make this `self.async_schedule_update_ha_state()`

Reply:
Actually, completely drop this. An update is called right after this.

https://github.com/home-assistant/home-assistant/blob/e122692b46715820dc4e56460d367a9fbc99b777/homeassistant/helpers/entity_platform.py#L257-L260